README.md

```diff
@@ -331,3 +331,210 @@ const MyArrayComponent = () => (Array(5).fill(<div />) as any) as JSX.Element;
 <!--START-SECTION:hooks-->
 
 ## Hooks
+
+Há suporte para Hooks em [`@types/react` a partir da versão v16.8](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a05cc538a42243c632f054e42eab483ebf1560ab/types/react/index.d.ts#L800-L1031).
+
+## useState
+
+Inferência automática de tipos funciona bem com valores simples
+
+```tsx
+const [val, toggle] = React.useState(false);
+//  infere-se que `val` é do tipo boolean
+// `toggle` aceita apenas booleans
+```
+
+Veja também no artigo em inglês (utilizando [Using Inferred Types](https://react-typescript-cheatsheet.netlify.app/docs/basic/troubleshooting/types/#using-inferred-types) se precisar usar um tipo complexo para o qual você depende da inferência.
+
+No entanto, muitos hooks são inicializados com valores nulos e você pode se perguntar como deve fazer para definir o tipo. Declare explicitamente o tipo e use um tipo de união (union type):
+
+```tsx
+const [user, setUser] = React.useState<IUser | null>(null);
+
+// mais tarde...
```

Comment:
Maybe we could have this as "mais adiante..."? What do you think?

Reply:
Sure, it's a better term.